File: year-planner.py
import datetime

class GoalPlanner:
    def __init__(self):
        self.goals = []

    def add_goal(self, goal, deadline):
        self.goals.append({"goal": goal, "deadline": deadline})

    def display_goals(self):
        print("\nYour Yearly Goals:")
        for i, goal in enumerate(self.goals, start=1):
            print(f"{i}. Goal: {goal['goal']}, Deadline: {goal['deadline']}")

    def run_planner(self):
        print("Welcome to the Yearly Goal Planner!")
        while True:
            print("\nChoose an option:")
            print("1. Add Goal")
            print("2. Display Goals")
            print("3. Exit")
            choice = input("Enter your choice (1/2/3): ")

            if choice == '1':
                goal = input("Enter your goal: ")
                deadline = input("Enter the deadline (YYYY-MM-DD): ")
                try:
                    deadline = datetime.datetime.strptime(deadline, "%Y-%m-%d").date()
                except ValueError:
                    print("Invalid date format. Please use YYYY-MM-DD.")
                    continue
                self.add_goal(goal, deadline)
                print("Goal added successfully!")

            elif choice == '2':
                self.display_goals()

            elif choice == '3':
                print("Exiting Yearly Goal Planner. Have a great year ahead!")
                break

            else:
                print("Invalid choice. Please enter 1, 2, or 3.")

if __name__ == "__main__":
    planner = GoalPlanner()
    planner.run_planner()

File: yearplanner.kt
import java.text.SimpleDateFormat
import java.util.*

data class Goal(val goal: String, val deadline: Date)

class GoalPlanner {
    private val goals = mutableListOf<Goal>()
    private val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.US)

    fun addGoal(goal: String, deadline: String) {
        try {
            val deadlineDate = dateFormat.parse(deadline)
            goals.add(Goal(goal, deadlineDate))
            println("Goal added successfully!")
        } catch (e: Exception) {
            println("Invalid date format. Please use YYYY-MM-DD.")
        }
    }

    fun displayGoals() {
        println("\nYour Yearly Goals:")
        goals.forEachIndexed { index, goal ->
            println("${index + 1}. Goal: ${goal.goal}, Deadline: ${dateFormat.format(goal.deadline)}")
        }
    }

    fun runPlanner() {
        println("Welcome to the Yearly Goal Planner!")
        while (true) {
            println("\nChoose an option:")
            println("1. Add Goal")
            println("2. Display Goals")
            println("3. Exit")
            print("Enter your choice (1/2/3): ")

            when (readLine()) {
                "1" -> {
                    print("Enter your goal: ")
                    val goal = readLine() ?: ""
                    print("Enter the deadline (YYYY-MM-DD): ")
                    val deadline = readLine() ?: ""
                    addGoal(goal, deadline)
                }
                "2" -> displayGoals()
                "3" -> {
                    println("Exiting Yearly Goal Planner. Have a great year ahead!")
                    break
                }
                else -> println("Invalid choice. Please enter 1, 2, or 3.")
            }
        }
    }
}

fun main() {
    val planner = GoalPlanner()
    planner.runPlanner()
}
